src/instructlab/model/dk_bench_utils.py

```diff
@@ -0,0 +1,320 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# pylint: disable=ungrouped-imports
+# Standard
+from datetime import datetime
+from typing import List
+import enum
+import logging
+import os
+import pathlib
+
+# Third Party
+from instructlab.eval.mt_bench_common import (
+    get_openai_client as get_local_openai_client,
+)
+from instructlab.eval.ragas import ModelConfig, RagasEvaluator
+from ragas.evaluation import EvaluationResult  # type: ignore
+import click
+import pandas as pd
+
+# Local
+from ..configuration import DEFAULTS
+from .evaluate import get_model_name as get_local_model_name
+from .evaluate import launch_server, validate_model
+
+logger = logging.getLogger(__name__)
+
+
+class IOFileType(enum.Enum):
+    CSV = "csv"
+    JSONL = "jsonl"
+    XLSX = "xlsx"
+
+
+def validate_output_file_formats(file_formats: List[str]):
+    for file_format in file_formats:
+        valid_formats = {item.value for item in IOFileType}
+        if file_format not in valid_formats:
+            raise ValueError(
+                f"File format {file_format} is not a valid output format. Format must be one of {valid_formats}"
+            )
+
+
+def create_results_file_name(
+    file_format: str, output_dir: str, timestamp: str, model_name: str
+) -> str:
+    if IOFileType.CSV.value == file_format:
+        file_type = "csv"
+    elif IOFileType.XLSX.value == file_format:
+        file_type = "xlsx"
+    elif IOFileType.JSONL.value == file_format:
+        file_type = "jsonl"
+    else:
+        raise ValueError("File format is not one of: csv, xlsx, jsonl")
+
+    output_path = pathlib.Path(output_dir)
+    if not output_path.exists() or not output_path.is_dir():
+        output_dir = DEFAULTS.EVAL_DATA_DIR
+        logger.debug(
+            "Output dir provided is not a directory or does not exist, writing results to default evaluation output directory %s",
+            output_dir,
+        )
+
+    return f"{output_dir}/responses-{model_name}-{timestamp}.{file_type}"
```

Comment:
is this an agreed upon file path format?

Reply:
I think it's sane, but I would revise it like this:


```suggestion
    return f"{output_dir}/{model-name}/responses-{timestamp}.{file_type}"
```